lib/index.js

```diff
@@ -284,10 +284,18 @@ FSWatcher.prototype._getWatchHelpers = function(path, depth) {
 // Returns the directory's tracking object
 FSWatcher.prototype._getWatchedDir = function(directory) {
   var dir = sysPath.resolve(directory);
+  var watcherRemove = this._remove.bind(this);
   if (!(dir in this._watched)) this._watched[dir] = {
     _items: Object.create(null),
     add: function(item) {this._items[item] = true;},
-    remove: function(item) {delete this._items[item];},
+    remove: function(item) {
+      delete this._items[item];
+      if (!this.children().length) {
+        fs.readdir(dir, function(err) {
+          if (err) watcherRemove(sysPath.dirname(dir), sysPath.basename(dir));
+        }.bind(this));
```

Comment:
oops that one isn't needed anymore... in an earlier iteration I hadn't aliased `this._remove`. Good catch